src/vsphere_cpi/spec/unit/cloud/vsphere/vm_config_spec.rb

```diff
@@ -20,6 +20,41 @@ def fake_datastore(name, free_space, mob = nil)
     let(:cluster_picker) { ClusterPicker.new(0, 0) }
     let(:cluster_provider) { nil }
 
+    describe '#upgrade_hw_version?' do
+      context 'when upgrade hardware version is nil (not specified) in both vm_type and global config' do
+        let(:cloud_properties) {{}}
+        let(:vm_type) { VmType.new(datacenter, cloud_properties)}
+        let(:input) { { vm_type: vm_type } }
```

Comment:
input is same for all 3 context, so it can be defined under describe '#upgrade_hw_version?' block